### PR TITLE
Use ImageJoinStreaming

### PR DESCRIPTION
**Pull request by erikbern**
* [X] Use `ImageJoinStreaming`
* [X] Use proper formatting / output management
* [X] Filter out image logs from app logs
* [ ] Get rid of the logs loop from `Stub._run` (will do later)

### CODE REVIEW

File: modal/image.py [hunk 1]
```diff
@@ -226,21 +226,27 @@ async def _load(resolver: Resolver, existing_object_id: str):
...
-                    raise RemoteError("Image build timed out. Please try again with a larger `timeout` parameter.")
-                elif response.result.status == api_pb2.GenericResult.GENERIC_STATUS_SUCCESS:
-                    break
-                else:
-                    raise RemoteError("Unknown status %s!" % response.result.status)
+            last_entry_id: Optional[str] = None
+            result: Optional[api_pb2.GenericResult] = None
+            while result is None:
+                request = api_pb2.ImageJoinStreamingRequest(image_id=image_id, timeout=55, last_entry_id=last_entry_id)
+                async for response in unary_stream(resolver.client.stub.ImageJoinStreaming, request):
```
[the remark this answers is not in the document]
Comment (aksh-at) [reply modal/image.py:233]: Perhaps we can generalize the logic in `get_logs_loop` that retries on GRPC errors: https://github.com/modal-labs/modal-client/blob/d33fba6dcd4dab6f60aee8813ccb7d4730ec74c2/modal/_output.py#L399

[the remark this answers is not in the document]
Comment (erikbern) [reply modal/image.py:233]: yeah maybe i can bake that into the `unary_stream` helper actually

[the remark this answers is not in the document]
Comment (erikbern) [reply modal/image.py:233]: will do that on the main branch as a separate PR

[the remark this answers is not in the document]
Comment (aksh-at) [reply modal/image.py:233]: I think there might be some uses of `unary_stream` that don't keep track of a `last_entry_id`. The retry behavior for the logs endpoints preserves the `last_entry_id` between requests.

[the remark this answers is not in the document]
Comment (erikbern) [reply modal/image.py:233]: actually it's hard to do that in a general way, i'll just roll it into this one

**Pull request merged**